src/state/insert-text.action.test.ts

```diff
@@ -142,6 +142,56 @@ describe('InsertTextAction', () => {
                     endOffset: 5
                 }
             }
+        },
+        {
+            name: 'should add 1 space at cursor position',
```

Comment:
Let's add a test case for adding multiple spaces so that we are confident that the bug is caused by rendering.

Reply:
Sure